cluster/cluster.go

```diff
@@ -563,3 +580,117 @@ func MustGetSegmentConfiguration(connection *dbconn.DBConn, getMirrors ...bool)
 	gplog.FatalOnError(err)
 	return segConfigs
 }
+
+/*GetSegmentConfigurationFromFile to parse gpsegconfig_dump file to retrieve segment configuration information.
```

Comment:
Suggested refactor:
```

// GetSegmentConfigurationFromFile parses the gpsegconfig_dump file to retrieve segment configuration information.
// Recommended use of the api is to get the contents of gp_segment_configuration when database is down.
// If the database is up, use GetSegmentConfiguration()/MustGetSegmentConfiguration() instead.
//  
// gpsegconfig_dump file gets created at $COORDINATOR_DATA_DIR/gpsegconfig_dump by the fts process. The frequency of
// writing to this file is governed by various fts gucs. Note: Since the gpsegconfig_dump file is updated by fts process the // information returned by this function can be a bit stale since user can configure fts to run less frequently
// 
// The gpsegconfig_dump file follows a structured format, as illustrated in the example below:
// 1 -1 p p n u 6000 localhost localhost /data/temp1
// 2 0 p p n u 6002 localhost localhost /data/temp2
// 3 1 p p n u 6003 localhost localhost /data/temp3
// 4 2 p p n u 6004 localhost localhost /data/temp4
//
// <Include an example with data dir and mention the gpdb version for each of these examples> 

// Input: Coordinator data dir path
// It returns the populated SegConfig object and error if any
```